source/blog/openshift-openId-integration-with-engine-sso.html.md

```diff
@@ -0,0 +1,129 @@
+
+Install Kibana/Elasticsearch/OpenShift on CentOS7 or RHEL 7.3 as described in https://github.com/ViaQ/Main/blob/master/README-mux.md
+
+## Installing oVirt Engine
+
+Setup oVirt Engine on a separate host ovirt-engine.example.com as described in https://www.ovirt.org/download/
+
+## Setting up oVirt Engine certificate on OpenShift machine
+
+Get the oVirt Engine CA as described here https://www.ovirt.org/documentation/how-to/guest-console/connect-to-spice-console-without-portal/
+
+```sh
+scp root@${OVIRT}:/etc/pki/ovirt-engine/ca.pem ${CA_FILE}
+```
+
+Add the certificate to system wide trusted certificates
+Copy certificate to /etc/pki/ca-trust/source/anchors/ and run update-ca-trust
+
+## Register a new sso client on ovirt-engine host
+
+Run the client registration tool ovirt-register-sso-client to register a new sso client. The tool will prompt the user to enter the client id, location of the client certificate (downloaded to oVirt engine host) and the callback url prefix. Make note of the client id and client secret generated by the tool. The client id and client secret need to be entered in the master configuration file on the OpenShift host to configure authentication with oVirt Engine.
+
+## Setup oauthconfig on Kibana/Elasticsearch/OpenShift host 
+
+On Kibana/Elasticsearch/OpenShift host edit /etc/origin/master/master-config.yaml to setup oauthconfig as below
+
+```yaml
+oauthConfig:
+  assetPublicURL: https://openshift.example.com:8443/console/
+  grantConfig:
+    method: auto
+  identityProviders:
+  - challenge: true
+    login: true
+    mappingMethod: claim
+    name: my_openid_connect
+    provider:
+      apiVersion: v1
+      kind: OpenIDIdentityProvider
+      clientID: <client id specified in previous step>
+      clientSecret: <client id generated in previous step>
+      extraScopes:
+      - ovirt-app-api
+      - ovirt-ext=auth:sequence-priority=~
+      extraAuthorizeParameters:
+        include_granted_scopes: "true"
+claims:
+        id:
+        - custom_id_claim
+        - sub
+        preferredUsername:
+        - preferred_username
+        - email
+        name:
+        - nickname
+        - given_name
+        - name
+        email:
+        - custom_email_claim
+        - email
+      urls:
+        authorize: https://ovirt-engine.example.com/ovirt-engine/sso/oauth/authorize
+        token: https://ovirt-engine.example.com/ovirt-engine/sso/oauth/token
+  masterCA: ca-bundle.crt
+  masterPublicURL: https://openshift.example.com:8443
+  masterURL: https://openshift.example.com:8443
+  sessionConfig:
+    sessionMaxAgeSeconds: 3600
+    sessionName: ssn
+    sessionSecretsFile: /etc/origin/master/session-secrets.yaml
+  tokenConfig:
+    accessTokenMaxAgeSeconds: 86400
+    authorizeTokenMaxAgeSeconds: 500
+```
+## Restart oVirt Engine
+
+```ssh
+service ovirt-engine restart
+```
+
+## Restart origin-master and origin-node on OpenShift host
+
+```ssh
+service origin-master restart
```

Comment:
Right, thanks, I missed this one, we should now use only systemctl to operate services